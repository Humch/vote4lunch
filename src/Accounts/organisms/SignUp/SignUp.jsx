import React from 'react';
import { connect } from 'react-redux';

import PropTypes from 'prop-types';

import { FormGroupSignUp, SignUpMessage } from '../..';

import './SignUp.scss';

const SignUp = ({ payload }) => (
  <div className="SignUp container-fluid">
    <div className="row no-gutters justify-content-center align-items-center h-100">
      <div
        className=" col-xs-12 col-md-6 col-lg-4 bg-blue p-5 rounded"
      >
        {
          payload.created
            ? (
              <SignUpMessage />
            )
            : (
              <FormGroupSignUp />
            )
          }
      </div>

    </div>
  </div>
);

SignUp.propTypes = {
  payload: PropTypes.shape({
    created: PropTypes.bool.isRequired
  }),
};

const mstp = ({ addUser: addUserReducer }) => ({
  payload: addUserReducer.payload,
});

export default connect(mstp)(SignUp);
