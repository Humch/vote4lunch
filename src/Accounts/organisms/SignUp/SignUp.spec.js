/* eslint-disable no-undef */
import React from 'react';
import reduxThunk from 'redux-thunk';
import { Provider } from 'react-redux';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import SignUp from './SignUp';

describe('SignUp Snapshot', () => {
  const initialState = {
    userData: {
      pseudo: '',
      email: '',
      password: '',
      passwordRepeater: '',
    },
    addUser: {
      payload: {
        created: false,
      },
      loading: false,
    },
    pseudoChecker: {
      payload: {
        exist: false,
      },
      loading: false,
    },
    emailChecker: {
      payload: {
        exist: false,
      },
      loading: false,
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of SignUp', () => {
    const renderedValue = renderer.create(
      <Provider store={store}>
        <SignUp />
      </Provider>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('SignUp Snapshot with loading', () => {
  const initialState = {
    userData: {
      pseudo: '',
      email: '',
      password: '',
      passwordRepeater: '',
    },
    addUser: {
      payload: {
        created: false,
      },
      loading: true,
    },
    pseudoChecker: {
      payload: {
        exist: false,
      },
      loading: false,
    },
    emailChecker: {
      payload: {
        exist: false,
      },
      loading: false,
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of SignUp with loading', () => {
    const renderedValue = renderer.create(
      <Provider store={store}>
        <SignUp />
      </Provider>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('SignUp Snapshot with user created', () => {
  const initialState = {
    userData: {
      pseudo: '',
      email: '',
      password: '',
      passwordRepeater: '',
    },
    addUser: {
      payload: {
        created: true,
      },
      loading: false,
    },
    pseudoChecker: {
      payload: {
        exist: false,
      },
      loading: false,
    },
    emailChecker: {
      payload: {
        exist: false,
      },
      loading: false,
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of SignUp with user created', () => {
    const renderedValue = renderer.create(
      <Provider store={store}>
        <SignUp />
      </Provider>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('SignUp (Shallow + passing the {store} directly)', () => {
  const initialState = {
    userData: {
      pseudo: 'mickey',
      email: 'mickey@mou.se',
      password: 'dingolefoldingo',
      passwordRepeater: 'dingolefoldingo',
    },
    addUser: {
      payload: {
        created: false,
      },
      loading: false,
    },
    pseudoChecker: {
      payload: {
        exist: false,
      },
      loading: false,
    },
    emailChecker: {
      payload: {
        exist: false,
      },
      loading: false,
    },
  };
  const mockStore = configureStore();
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<SignUp store={store} />);
  });

  it('render the connected(SMART) component', () => {
    expect(container.length).toEqual(1);
  });

  it('check Prop matches with initialState', () => {
    expect(container.prop('pseudo')).toEqual(initialState.userData.pseudo);
    expect(container.prop('email')).toEqual(initialState.userData.email);
    expect(container.prop('password')).toEqual(initialState.userData.password);
    expect(container.prop('passwordRepeater')).toEqual(initialState.userData.passwordRepeater);
    expect(container.prop('payload')).toEqual(initialState.addUser.payload);
    expect(container.prop('loading')).toEqual(initialState.addUser.loading);
  });
});

describe('SignUp - REACT-REDUX (Mount + wrapping in <Provider>)', () => {
  const initialState = {
    userData: {
      pseudo: 'mickey',
      email: 'mickey@mou.se',
      password: 'dingolefoldingo',
      passwordRepeater: 'dingolefoldingo',
    },
    addUser: {
      payload: {
        created: false,
      },
      loading: false,
    },
    pseudoChecker: {
      payload: {
        exist: false,
      },
      loading: false,
    },
    emailChecker: {
      payload: {
        exist: false,
      },
      loading: false,
    },
  };
  const middlewares = [reduxThunk];
  const mockStore = configureStore(middlewares);
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    store.clearActions();

    wrapper = mount(
      <Provider store={store}>
        <SignUp />
      </Provider>
    );
  });

  it('render the connected(SMART) component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('check if form is submit', () => {
    const signUpForm = wrapper.find('form');
    signUpForm.simulate('submit', { preventDefault: jest.fn() });
    const action = store.getActions();

    expect(action[0].type).toBe('ADD_USER_BEGIN');
  });
});
