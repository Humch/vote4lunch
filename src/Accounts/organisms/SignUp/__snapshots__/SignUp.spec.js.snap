// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Snapshot capturing Snapshot of SignUp 1`] = `
<div
  className="SignUp container-fluid"
>
  <div
    className="row no-gutters justify-content-center align-items-center h-100"
  >
    <div
      className=" col-xs-12 col-md-6 col-lg-4 bg-blue p-5 rounded"
    >
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            className=" text-white"
            htmlFor="pseudo"
          >
            Choisis ton pseudo
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              id="pseudo"
              name="pseudo"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            
            <div
              className="invalid-feedback"
            >
              Ce pseudo existe déjà
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            className=" text-white"
            htmlFor="email"
          >
            Indique ton email
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              id="email"
              name="email"
              onChange={[Function]}
              required={true}
              type="email"
              value=""
            />
            
            <div
              className="invalid-feedback"
            >
              Cet email est déjà utilisé
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            className=" text-white"
            htmlFor="password"
          >
            Choisis ton mot de passe
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              id="password"
              name="password"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            
            
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            className=" text-white"
            htmlFor="passwordRepeater"
          >
            Répète ton mot de passe
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              id="passwordRepeater"
              name="passwordRepeater"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            
            <div
              className="invalid-feedback"
            >
              Les mots de passe sont différents
            </div>
          </div>
        </div>
        <div
          className="text-center mt-5"
        >
          <button
            aria-label={null}
            className="btn btn-success"
            onClick={[Function]}
          >
            Valider
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`SignUp Snapshot with loading capturing Snapshot of SignUp with loading 1`] = `
<div
  className="SignUp container-fluid"
>
  <div
    className="row no-gutters justify-content-center align-items-center h-100"
  >
    <div
      className=" col-xs-12 col-md-6 col-lg-4 bg-blue p-5 rounded"
    >
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            className=" text-white"
            htmlFor="pseudo"
          >
            Choisis ton pseudo
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              id="pseudo"
              name="pseudo"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            
            <div
              className="invalid-feedback"
            >
              Ce pseudo existe déjà
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            className=" text-white"
            htmlFor="email"
          >
            Indique ton email
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              id="email"
              name="email"
              onChange={[Function]}
              required={true}
              type="email"
              value=""
            />
            
            <div
              className="invalid-feedback"
            >
              Cet email est déjà utilisé
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            className=" text-white"
            htmlFor="password"
          >
            Choisis ton mot de passe
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              id="password"
              name="password"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            
            
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            className=" text-white"
            htmlFor="passwordRepeater"
          >
            Répète ton mot de passe
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              id="passwordRepeater"
              name="passwordRepeater"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            
            <div
              className="invalid-feedback"
            >
              Les mots de passe sont différents
            </div>
          </div>
        </div>
        <div
          className="text-center mt-5"
        >
          <button
            aria-label={null}
            className="btn btn-success"
            onClick={[Function]}
          >
            Requête en cours
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`SignUp Snapshot with user created capturing Snapshot of SignUp with user created 1`] = `
<div
  className="SignUp container-fluid"
>
  <div
    className="row no-gutters justify-content-center align-items-center h-100"
  >
    <div
      className=" col-xs-12 col-md-6 col-lg-4 bg-blue p-5 rounded"
    >
      <div>
        <p
          className="text-center text-white"
        >
          Ton compte a été créé.
        </p>
        <p
          className="text-center text-white"
        >
          Tu peux maintenant t'identifier.
        </p>
        <p
          className="text-center"
        >
          <a
            aria-label={null}
            className="mr-2 btn btn-success"
            href="/signin"
            id="loginButton"
            onClick={[Function]}
          >
            S'identifier
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
